tests/integration/sentinel/transitions/create-user-for-contacts.spec.js

```diff
@@ -255,288 +174,620 @@ describe('create_user_for_contacts', () => {
-    const missingNamePattern = /Replacement contact \[new_person] must have a name\./;
-    const newPerson = { ...NEW_PERSON, name: undefined };
+    it('adds user for new contact with multiple roles', async () => {
+      await utils.updateSettings(getSettings(), 'sentinel');
 
-    const collectLogs = await utils.collectSentinelLogs(missingNamePattern);
-    await utils.updateSettings(getSettings(), 'sentinel');
-    await utils.createUsers([ORIGINAL_USER]);
-    newUsers.push(ORIGINAL_USER.username);
-    await utils.saveDoc(newPerson);
-    const originalContact = await utils.getDoc(ORIGINAL_PERSON._id);
-    originalContact.user_for_contact = {
-      replace: {
-        [ORIGINAL_USER.username]: {
-          replacement_contact_id: newPerson._id,
-          status: 'READY'
-        }
-      },
-    };
-    await utils.saveDoc(originalContact);
-    await sentinelUtils.waitForSentinel(originalContact._id);
-    const { transitions } = await sentinelUtils.getInfoDoc(originalContact._id);
+      const originalContact = Object.assign(
+        {
+          roles: ['chw', 'other-role'],
+          user_for_contact: { add: { status: 'READY' }, }
+        },
+        NEW_PERSON
+      );
+      await utils.saveDoc(originalContact);
+      await sentinelUtils.waitForSentinel(originalContact._id);
+      const { transitions } = await sentinelUtils.getInfoDoc(originalContact._id);
+
+      // Transition successful
+      assert.isTrue(transitions.create_user_for_contacts.ok);
+      // Contact updated
+      const updatedContact = await utils.getDoc(originalContact._id);
+      assert.equal(updatedContact.user_for_contact.add.status, 'COMPLETE');
+
+      // New user created
+      const [newUserSettings, ...additionalUsers] = await utils.getUserSettings({ contactId: originalContact._id });
+      assert.isEmpty(additionalUsers);
+      newUsers.push(newUserSettings.name);
+      assert.deepInclude(newUserSettings, {
+        roles: originalContact.roles,
+        phone: originalContact.phone,
+        facility_id: originalContact.parent._id,
+        contact_id: originalContact._id,
+        fullname: originalContact.name,
+      });
+      assert.isTrue(newUserSettings.token_login.active);
+      assert.match(newUserSettings._id, /^org\.couchdb\.user:new-person-\d\d\d\d/);
+      assert.match(newUserSettings.name, /^new-person-\d\d\d\d$/);
+      // Login token sent
+      const queuedMsgs = await getQueuedMessages();
+      assert.lengthOf(queuedMsgs, 1);
+      assert.deepInclude(queuedMsgs[0], {
+        type: 'token_login',
+        user: newUserSettings._id
+      });
+      assert.equal(queuedMsgs[0].tasks[0].messages[0].to, NEW_PERSON.phone);
+    });
 
-    assert.isFalse(transitions.create_user_for_contacts.ok);
-    assert.isNotEmpty(await collectLogs());
-    await expectError(missingNamePattern);
-  });
+    // TODO Need to add support for this scenario
+    it.skip('adds user for new contact with single role', async () => {
```

Comment:
In this scenario, the transition falls back to `user_for_contact.add.roles` which should be set to the role that was selected when creating the contact like [the client-side transition does](https://github.com/medic/cht-core/blob/7753-supervisor-chw-create/webapp/src/ts/services/create-user-for-contacts.service.ts#L117)